src/Stratis.Bitcoin/P2P/PeerAddressManager.cs

```diff
@@ -148,55 +163,45 @@ public void PeerAttempted(IPEndPoint endpoint, DateTime peerAttemptedAt)
         public void PeerConnected(IPEndPoint endpoint, DateTimeOffset peerConnectedAt)
         {
             PeerAddress peer = this.FindPeer(endpoint);
-            if (peer == null)
-                return;
 
-            peer.SetConnected(peerConnectedAt);
+            peer?.SetConnected(peerConnectedAt);
         }
 
         /// <inheritdoc/>
         public void PeerDiscoveredFrom(IPEndPoint endpoint, DateTime peerDiscoveredFrom)
         {
             PeerAddress peer = this.FindPeer(endpoint);
-            if (peer == null)
-                return;
 
-            peer.SetDiscoveredFrom(peerDiscoveredFrom);
+            peer?.SetDiscoveredFrom(peerDiscoveredFrom);
         }
 
         /// <inheritdoc/>
         public void PeerHandshaked(IPEndPoint endpoint, DateTimeOffset peerHandshakedAt)
         {
             PeerAddress peer = this.FindPeer(endpoint);
-            if (peer == null)
-                return;
 
-            peer.SetHandshaked(peerHandshakedAt);
+            peer?.SetHandshaked(peerHandshakedAt);
         }
 
         /// <inheritdoc/>
         public void PeerSeen(IPEndPoint endpoint, DateTime peerSeenAt)
         {
             PeerAddress peer = this.FindPeer(endpoint);
-            if (peer == null)
-                return;
 
-            peer.SetLastSeen(peerSeenAt);
+            peer?.SetLastSeen(peerSeenAt);
         }
 
         /// <inheritdoc/>
         public PeerAddress FindPeer(IPEndPoint endPoint)
         {
-            KeyValuePair<IPEndPoint, PeerAddress> peer = this.peers.Skip(0).SingleOrDefault(p => p.Key.Match(endPoint));
-            if (peer.Value != null)
-                return peer.Value;
-            return null;
+            KeyValuePair<IPEndPoint, PeerAddress> peer = this.peerInfoByPeerAddress.Skip(0).SingleOrDefault(p => p.Key.Match(endPoint));
+            return peer.Value;
```

Comment:
This wrong, peer can be null here ;)

Reply:
Nope, KeyValuePair is a structure, not a class, so can't be null

Reply:
Of course :)